packages/hardhat/contracts/QuadraticDiplomacyContract.sol

```diff
@@ -31,6 +35,31 @@ contract QuadraticDiplomacyContract is Distributor, AccessControl {
         _;
     }
 
+    // Warning: Only use in combination with onlyAdmin. The caller will get any leftovers.
+    // Warning: Only ETH and the tokens from the provided token address are reimbursed.
+    modifier startNewElectionAfter(address tokenAddress) {
+        _;
+
+        // remove all voter roles
+        // an alternative (and less gas heavy?) way to do this would be to change the VOTER_ROLE to keccak256(abi.encodePacked(block.number));
+        for (uint256 i = 0; i < getRoleMemberCount(VOTER_ROLE); i++) {
+            revokeRole(VOTER_ROLE, getRoleMember(VOTER_ROLE, i));
+        }
+
+        // if there are leftover ETH in the contract, send it to the admin
+        if (address(this).balance > 0) {
+            payable(msg.sender).transfer(address(this).balance);
+        }
+
+        // if there are leftover tokens in the contract, send it to the admin
+        if (tokenAddress != address(0)) {
+            IERC20(tokenAddress).transfer(msg.sender, IERC20(tokenAddress).balanceOf(address(this)));
+        }
```

Comment:
This is a nice touch! 

Personally, this is one of the things I wouldn't have implemented until it's requested, because maybe admins prefer to save the amount in the contract for future elections.

Reply:
Question: is `transfer` the way to go when sending ETH? I've also seen contracts using `safeTransfer` or `.call{value: msg.value}("")`

Reply:
> This is a nice touch!
> 
> Personally, this is one of the things I wouldn't have implemented until it's requested, because maybe admins prefer to save the amount in the contract for future elections.

That makes sense, as the admin stays the same. Should I create a PR removing it?

Reply:
> Question: is `transfer` the way to go when sending ETH? I've also seen contracts using `safeTransfer` or `.call{value: msg.value}("")`

Thats true, Should have used `call{value: msg.value}` for transferring ETH and `safeTransfer` for ERC20 👍

Reply:
@carletex, should I create a PR removing it or fixing it?

Reply:
@sogasg If you remove it we can always go back to this PR and get it back... and If you fix it and keep it we can always remove it :D So, up to you.

Reply:
Removing it for now: https://github.com/moonshotcollective/q-dip/pull/22 ;)